worker/sshserver/server_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver_test
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"net"
+	"time"
+
+	"github.com/juju/testing"
+	"golang.org/x/crypto/ssh"
+	"google.golang.org/grpc/test/bufconn"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/worker/sshserver"
+)
+
+type sshServerSuite struct {
```

Comment:
More thorough testing should be added, specially around the tcp handlers which are quite complex.

Reply:
For now they don't do anything, but as we add in the auth, forwarding, unit resolution etc, these tests will grow. This initial test is really just to show "hey look, it routes through a jump correctly to the in-memory server".